visualization/include/pcl/visualization/impl/pcl_visualizer.hpp

```diff
@@ -658,7 +658,7 @@ pcl::visualization::PCLVisualizer::addText3D (
 
   // check all or an individual viewport for a similar id
   rens_->InitTraversal ();
-  for (size_t i = viewport; rens_->GetNextItem () != NULL; ++i)
+  for (size_t i = viewport; rens_->GetNextItem () != nullptr; ++i)
```

Comment:
Well, I started this changes, but stopped, because changes would be reverted by [readability-implicit-bool-conversion](https://clang.llvm.org/extra/clang-tidy/checks/readability-implicit-bool-conversion.html). In general this code looks really bad, because value of next item is not used, so `GetNumberOfItems` should be used.

Reply:
> Well, I started this changes, but stopped, because changes would be reverted by [readability-implicit-bool-conversion](https://clang.llvm.org/extra/clang-tidy/checks/readability-implicit-bool-conversion.html). 

Is this a clang-tidy point we want to adopt? I have a weak preference for no. 

> In general this code looks really bad, because value of next item is not used, so `GetNumberOfItems` should be used.

I see your point. Oh well, there's never lack of things to do around here.

Reply:
I'm on Sergio's side here. As far as I can see, the check in question has a _AllowPointerConditions_ option, so we can adopt it without having to have `== nullptr` everywhere.

Reply:
You are right @taketwo: Didn't saw the option to disable it for `nullptr`.